plugins/webhooks/helper.js

```diff
@@ -661,23 +663,39 @@ async function batchStopJobs({ pipelines, prNum, action, name }) {
  * @param  {Hapi.h}       h                     Response toolkit
  */
 async function pullRequestOpened(options, request, h) {
-    const { hookId } = options;
+    const { hookId, username, scmConfig, pipeline } = options;
+    const { scmContext } = scmConfig;
 
-    return createPREvents(options, request)
-        .then(events => {
-            events.forEach(e => {
-                request.log(['webhook', hookId, e.id], `Event ${e.id} started`);
-            });
+    try {
+        const events = await createPREvents(options, request);
+        const hasBuildEvents = events.filter(e => e.builds !== null);
 
-            return h.response().code(201);
-        })
-        .catch(err => {
-            logger.error(
-                `Failed to pullRequestOpened: [${hookId}, pipeline:${options.pipeline && options.pipeline.id}]: ${err}`
-            );
+        for (const event of hasBuildEvents) {
+            const virtualJobBuilds = event.builds.filter(b => b.status === 'CREATED');
```

Comment:
is there a more explicit check that can be used on the build/job to denote that it's a virtual Job Build ?
Would there be any other scenario for `CREATED` build status

Reply:
Thank you, I fixed.